content/pixel_fifo.md

```diff
@@ -0,0 +1,252 @@
+# Pixel FIFO
+Before we get started, all references to a cycle are meant as T-cycles
+(4.19 MHz) and cycle counts are doubled on CGB in double speed mode. Also
+when stating that a certain action "lengthens mode 3" this means that
+mode 0 (hblank) is shortened. The extra time has to be made up somewhere
+and the lengthening of mode 3 eats into the mode that comes afterwards,
+which happens to be mode 0 as shown in the following diagram.
+
+![](imgs/game-boy-lcd-refresh-diagram.png "imgs/game-boy-lcd-refresh-diagram.png")
+
+## FIFO info
+FIFO stands for First In, First Out. The first pixel to be pushed to the
+FIFO is the first pixel to be popped off. In theory that sounds great,
+in practice there are a lot of intricacies.
+
+There are two pixel FIFOs. One for background pixels and one for OAM
+(sprite) pixels. These two FIFOs are not shared. They are independent
+of each other. The two FIFOs are mixed only when popping items. Sprites
+take priority unless they're transparent (color 0) which will be
+explained in detail later. The FIFOs can each hold up to 16 pixels at a
+time. The two FIFOs are manipulated only during mode 3 (pixel transfer)
+and they are both cleared at the beginning of mode 3.
+
+Each pixel in the FIFO has four properties:
+- Color: a value between 0 and 3
+- Palette: on CGB a value between 0 and 7 and on DMG a value between 1 and 0
+- Sprite Priority: on CGB this is the OAM index for the sprite and on DMG this is 0
+- Background Priority: holds the value of the [OBJ-to-BG Priority](#vram-sprite-attribute-table-oam) bit
+
+## FIFO Pixel Fetcher
```

Comment:
How about starting with "Each Pixel FIFO contains a Pixel Fetcher, this fetcher runs in parallel to the FIFO and is responsible for fetching the next tile's data for the FIFO to draw."